app/views/course/video/submission/submissions/edit.json.jbuilder

```diff
@@ -1,3 +1,8 @@
 json.video do
   json.videoUrl @video.url
 end
+
+json.discussion do
+  json.partial! 'course/video/topics/topics', locals: { topics: @topics }
+  json.partial! 'course/video/topics/posts', locals: { posts: @posts }
```

Comment:
Any particular reason why you prefer this method over providing the post partials within the topic partial instead?

Reply:
It was basically to mirror the the redux state for the frontend, so I don't have to flatten them out again.

The state is stored in a somewhat normalized form, with posts and topics each having their own map. This form allows each frontend component to be in charge of one entity. It also makes the reducers cleaner, since we can slice out the topic/post/replies to a function each.

The main problem I forsee is the sorting and filtering getting a bit heavy. I was looking into `reactjs/reselect` as recommended by the redux documentation for that.

Reply:
Thanks for the explanation. We can kiv the heavy sorting and filtering for now, but it will be something we look at to reduce some front-end load.